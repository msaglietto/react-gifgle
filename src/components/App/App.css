.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 50px;
}

.App-header {
  background-color: #222;
  height: 50px;
  padding: 10px;
  color: white;
  display: flex;
  align-items: center;
}

.App-title {
  font-size: 1.5em;
}

.App-SearchBar {
  margin-left: 20px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
