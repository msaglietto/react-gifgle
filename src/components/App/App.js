import React, { Component } from 'react'
import PropTypes from 'prop-types'

import SearchBar from '../SearchBar'
import ImageList from '../ImageList'

import logo from './logo.svg'
import './App.css'

class App extends Component {
  static displayName = 'App'

  static propTypes = {
    images: PropTypes.array.isRequired
  }

  state = {
    query: '',
    offset: 0,
    page: 1
  }

  componentWillMount () {
    this.props.getImages()
  }

  handleNewSearch = (query) => {
    this.props.getImages(query)
    this.setState({ query })
  }

  render () {
    return (
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h1 className='App-title'>Gifgle</h1>
          <div className='App-SearchBar'>
            <SearchBar onNewSearch={this.handleNewSearch} />
          </div>
        </header>
        <h2>
          You are seeing the results of {this.state.query || 'trending gif'}
        </h2>
        <ImageList images={this.props.images} />
      </div>
    )
  }
}

export default App
