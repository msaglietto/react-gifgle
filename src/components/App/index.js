import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { getImages } from '../../actions/images'

import App from './App'

const mapStateToProps = state => ({
  images: state.images.data
})

const mapDispatchToProps = dispatch => bindActionCreators({
  getImages
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App)
