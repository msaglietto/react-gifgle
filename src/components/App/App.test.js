/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

it('renders without crashing', () => {
  const props = {
    getImages: () => {},
    images: [],
    page: 0,
    total: 10
  }
  const wrapper = shallow(<App {...props} />)

  expect(wrapper).toMatchSnapshot()
})
