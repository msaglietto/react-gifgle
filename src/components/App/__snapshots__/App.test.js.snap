// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Gifgle
    </h1>
    <div
      className="App-SearchBar"
    >
      <SearchBar
        onNewSearch={[Function]}
      />
    </div>
  </header>
  <h2>
    You are seeing the results of 
    trending gif
  </h2>
  <ImageList
    images={Array []}
  />
</div>
`;
